### ISSUE TITLE
Errors upgrading 0.8.x to latest Cacti due to incorrectly detected data source profile id

### ISSUE THREAD

**Issue by MrDibbley** [Cacti/cacti#3319]
same as issue #1002  

The database upgrade fails. Here are the failed statements (and internal db version). Everything goes ok till upgrade to db version 1.0.5

version 1.0.5
[Fail]
UPDATE data_template_data SET data_source_profile_id = WHERE data_source_profile_id = 0

Here, obviously, there is stg missing before the WHERE. All other errors (below) seem legit syntax.

![image](https://user-images.githubusercontent.com/20700791/75881014-464d8b80-5e16-11ea-834b-fb812c0dfb4c.png)

set setmode as https://github.com/Cacti/cacti/issues/1002#issuecomment-357929994 and retested same result?

Thanks

**Comment by netniV** [comment]
Do me a favour, run the following:
```sql
SELECT id FROM data_source_profiles ORDER BY `default` DESC LIMIT 1
```

**Comment by MrDibbley** [comment]
SELECT id FROM data_source_profiles ORDER BY `default` DESC LIMIT 1;
Empty set (0.001 sec)

data_source_profiles never existed in the original 0.8 db

**Comment by netniV** [comment]
That should have been inserted by an earlier upgrade though (1.0.0 I would imagine).

**Comment by netniV** [comment]
OK, so looking at the v1.0.0 upgrade function, it attempts to look at `data_template_data_rra` if it exists.  Does that exist on your 0.8.8h database?

**Comment by netniV** [comment]
If possible, I would like a dump of the 0.8.8 database emailed to developers@cacti.net so I can do a proper step through and see occurs.

**Comment by MrDibbley** [comment]
yes data_template_data_rra exists in the 0.8 db, ill email you a mysqldump of the 0.8 db too.

**Comment by netniV** [comment]
Thanks, I'll take a look tonight.

**Comment by cigamit** [comment]
This was resolved as of the merging of the fetch branch into the 1.2.x branch and will be resolved as a part of the 1.2.11 release.  In the mean time, just download the 1.2.x branch ad restart the upgrade.
